### DISCUSSION TITLE
Reinvest dividend

### DISCUSSION THREAD

**Discussion by arp2164** [moneymanagerex/moneymanagerex#5168] [Q&A]
Can someone show me how to add dividend reinvestment in stock portfolio.
There are only two inputs 1. Withdrawal and 2. Deposit.
Adding transaction with deposit it removes the shares.
Do I have to use withdrawal?
Any help will be appreciated.
Ashok

**Comment by Elroban85** [comment]
Based on the video https://youtu.be/UJr2M743p10 @ 6:16 you can add a transaction without mentioning the shares.
So actually a deposit of dividends only.
Then you again would use the withdrawal to 'buy' the new shares.
I'm personally puzzling which category to use for the buying with the dividend's money cause including it with the normal purchase share category gives me the idea that I have invested more than I actually did. I'm leaning towards using the same category for both the withdrawal as the deposit to balance them out. 

The way the withdrawl and deposit work is also that they impact the total cost of the investment as the withdrawal increases the total cost but the deposit doesn't decrease. It would be nice if you could see the dividends in your stock portfolio as a separate column or have a preference to not include dividend reinvestments in the calculation of the total cost. Although I can understand that it is seen as an actual investment / cost as if you wouldn't have reinvested it, the money would have been income only.

**Comment by arp2164** [comment]
Yes you could do a transaction for $deposit without adding shares, which means one got cash dividend and not reinvested.
Currently I have created a cash (wallet) account for the brokerage. When dividends are given and reinvested, I manually add cash money, then transfer from this to shares/Mf and then buy new shares from transferred money. This way withdrawal/ deposit are balanced.
Yes it would be better if there is extra column to show dividends. This way normal purchases of shares and reinvested shares are clearly defined.
I hope that in future updates this could be added.

**Comment by n-stein** [comment]
Here are a few tips that might help:

There is a difference between a normal "checking" transaction and a "stock" transaction, and they are added in different ways. A "checking" transaction has no impact on the stock (shares, cost, avg price, etc) while a "stock" transaction does. You need one of each for dividend reinvestment, and you can do both of these directly in the share account with no need to have a "cash" account. Here's how I normally do this:

1. Create a deposit _checking_ transaction directly into the share account. This is done by clicking on the company name under the investment account (circled in red below), then clicking "New" at the bottom of the panel or right clicking and selecting "New Deposit" just as you would in any normal bank account. Enter the date, $ amount of the dividend, payee, and category. 

![image](https://user-images.githubusercontent.com/16599929/200992058-19c2798d-328d-41fd-ae8a-35e1d6c67c83.png)

2. Now create a withdrawal _stock_ transaction on the share account for the amount of shares purchased. This part is done within the investment portfolio view (again circled in red below, the parent investment account of the previous transaction) by right clicking on the stock and selecting "Add stock transaction" as shown. Here you put in the shares purchased and the price per share. Make sure type is "Withdrawal", the account is the share account, and the total amount matches the number from the first transaction.

![image](https://user-images.githubusercontent.com/16599929/200992955-45cf51f1-4849-4a29-a3f8-ecf3495bf605.png)
![image](https://user-images.githubusercontent.com/16599929/200993305-dd957b9c-f6bc-46f5-9fa6-3d5adde2c9d6.png)

Your total cost will increase by the total $ of the dividend, your avg share price will be the new weighted average price, and your share account balance change will be net-0 since you deposited & withdrew the same amount.

Example before:
Share account
![image](https://user-images.githubusercontent.com/16599929/200994147-6492fec1-1e9e-4ff2-b830-f2d946b14ca6.png)
Investment account
![image](https://user-images.githubusercontent.com/16599929/200994586-a65ee32f-f2ab-4875-8eed-99f9c5a515dc.png)

Example after:
Share account
![image](https://user-images.githubusercontent.com/16599929/200994730-dfd6a62d-f3fe-4c0e-abd7-c100c1e3d407.png)
Investment account
![image](https://user-images.githubusercontent.com/16599929/200994814-9e544f45-3832-46a4-ad3f-45dda1dc85b8.png)

**Reply by arp2164** [reply]
Thank you for your detailed report. I will now do per your method. I created A cash account but now it's not Neccessry thanks again for taking time in writing report
